ballerina/sql_client.bal

```diff
@@ -112,15 +116,37 @@ public isolated client class SQLClient {
     # + rowType - The type description of the entity to be retrieved
     # + fields - The fields to be retrieved
     # + include - The associations to be retrieved
+    # + whereClause - The `WHERE` clause of the query
+    # + orderByClause - The `ORDER BY` clause of the query
+    # + limitClause - The `LIMIT` clause of the query
+    # + groupByClause - The `GROUP BY` clause of the query
     # + return - A stream of records in the `rowType` type or a `persist:Error` if the operation fails
-    public isolated function runReadQuery(typedesc<record {}> rowType, string[] fields = [], string[] include = [])
-    returns stream<record {}, sql:Error?>|persist:Error {
+    public isolated function runReadQuery(typedesc<record {}> rowType, string[] fields = [], string[] include = [],
+                        sql:ParameterizedQuery whereClause = ``, sql:ParameterizedQuery orderByClause = ``,
+                        sql:ParameterizedQuery limitClause = ``, sql:ParameterizedQuery groupByClause = ``)
+                        returns stream<record {}, sql:Error?>|persist:Error|error {
         sql:ParameterizedQuery query = self.getSelectQuery(fields);
-
         foreach string joinKey in self.getJoinFields(include) {
             query = sql:queryConcat(query, check self.getJoinQuery(joinKey));
         }
-
+        if (whereClause.strings.length() != 0) {
+            query = sql:queryConcat(query, ` WHERE `, whereClause);
+        }
+        if (groupByClause.strings.length() != 0) {
+            query = addClauseToQuery(query,  groupByClause, ` GROUP BY `);
+        }
+        if (orderByClause.strings.length() != 0) {
+            query = addClauseToQuery(query, orderByClause, ` ORDER BY `);
+        }
+        if (limitClause.strings.length() != 0) {
+            if (limitClause.insertions.length() != 0) {
+                string queryInString = "LIMIT " + limitClause.strings[0] + limitClause.insertions[0].toString();
+                query = sql:queryConcat(query, stringToParameterizedQuery(queryInString));
+            } else {
+                query = sql:queryConcat(query, ` LIMIT `, limitClause);
+            }
```

Comment:
Isn't this logic covered in `addClauseToQuery()`

Reply:
No, the Limit clause has a single value. So we can convert it directly.